build.gradle

```diff
@@ -364,56 +371,54 @@ project(':iceberg-pig') {
 }
 
 // the runtime jar is a self-contained artifact for testing in a notebook
-project(':iceberg-runtime') {
+project(':iceberg-spark-runtime') {
   apply plugin: 'com.github.johnrengelman.shadow'
 
-  configurations {
-    shadow
-    compileOnly.extendsFrom shadow
-  }
-
   shadowJar {
-    configurations = [project.configurations.shadow]
     from(project.sourceSets.main.output)
   }
 
   tasks.build.dependsOn tasks.shadowJar
   tasks.install.dependsOn tasks.shadowJar
   tasks.javadocJar.dependsOn tasks.shadowJar
 
-  dependencies {
-    shadow project(':iceberg-api')
-    shadow project(':iceberg-common')
-    shadow project(':iceberg-core')
-    shadow project(':iceberg-parquet')
-    shadow project(':iceberg-spark')
-    shadow project(':iceberg-pig')
-    shadow project(':iceberg-hive')
-
-    shadow "org.apache.avro:avro"
-    shadow "org.apache.parquet:parquet-avro"
+  configurations {
+    compile {
+      // included in Spark
+      exclude group: 'org.slf4j'
+      exclude group: 'org.apache.commons'
+      exclude group: 'commons-pool'
+      exclude group: 'commons-codec'
+      exclude group: 'org.xerial.snappy'
+      exclude group: 'javax.xml.bind'
+    }
   }
 
-  publishing {
-    publications {
-      withType(IvyPublication) {
-        configurations {
-          'shadow' { extend 'runtime' }
-        }
-      }
-    }
+  dependencies {
+    compile project(':iceberg-spark')
   }
 
   shadowJar {
```

Comment:
I tested this out by publishing this jar locally*. It seems that together with the fat jar I see some of its transitive *project* dependencies are being pulled in. See
<img width="531" alt="Screen Shot 2019-07-05 at 6 28 48 AM" src="https://user-images.githubusercontent.com/129474/60725553-3a5ba380-9eee-11e9-932b-49af7f795bca.png">

Not sure what is a good way to fix this, might require some Gradle magic. A workaround could be to use `transitive = false` when depending on iceberg-spark-runtime, until we fix this.

Note: I published the jar locally by adding the maven plugin (`plugins { id 'maven'}) and then running the command `./gradlew clean install

Reply:
It is a good idea to clean up the declared dependencies, but this Jar is not intended to be pulled in as a dependency. It is intended to be a single Jar you put in Spark's `jars` folder to add Iceberg support.

Reply:
I was thinking of Spark's `--packages` option, but I take your point

Reply:
Actually, @rdblue , why do you prefer this tool's deployment mechanism as a jar in jars/ folder? 

If you do --packages option, you don't have to manage this because the spark shell excludes spark things from resolving: https://github.com/apache/spark/blob/a950570f91db56cbae488c82def49cd0da16e996/core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala#L1195

Reply:
I'm fine making sure this works with `--packages`, but that's not sufficient. Since we don't have a release yet, we can't use `--packages` if it needs to pull from maven central. Right now, users just need to build the project and add one Jar. This will also work for people that want to test out new features between releases.